src/cryptonote_basic/miner.cpp

```diff
@@ -763,45 +784,54 @@ namespace cryptonote
     return (uint8_t)( ceil( (other * 1.f / total * 1.f) * 100) );    
   }
   //-----------------------------------------------------------------------------------------------------    
-  bool miner::on_battery_power()
+  boost::logic::tribool miner::on_battery_power()
```

Comment:
I think the tribool is ok, but my c++ code is poop (maybe all my code, lol), so if you think it's better I'll implement it.